### PR TITLE
[freebox] Hot fix for login to the freebox

### PR DESCRIPTION
**Pull request by lolodomo**
PR #7460 broke the freebox binding

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by cpmeister** [comment]
Could you add a unit test to this as well? I'd like to prevent any future mishaps.

**Comment by lolodomo** [comment]
I don't know where @cweitkamp put its unit test.
We cannot wait too much, the binding is just unusable until this patch is applied for any user that has migrated to OH 2.5.5 !

**Comment by cweitkamp** [comment]
See https://github.com/openhab/openhab-addons/pull/7460#issuecomment-631025326.

**Comment by lolodomo** [comment]
I will not take a long time to introduce all the testing stuff to the binding (I am not enough familiar with that) even more when the bug was not introduced by me, 
The priority is to provide a fix to end users as fast as possible.

**Comment by cpmeister** [comment]
Just add this file and fill in the blanks:
`src/test/java/org/openhab/binding/freebox/internal/api/FreeboxApiManagerTest`
```
package org.openhab.binding.freebox.internal.api;

import org.eclipse.jdt.annotation.NonNullByDefault;
import org.junit.Assert;
import org.junit.Test;

@NonNullByDefault
public class FreeboxApiManagerTest{

    @Test
    public void hmacSha1Test() throws Exception {
        String expected = "_______";
        String actual = FreeboxApiManager.hmacSha1("_______", "_______");
        Assert.assertEquals(expected, actual);
    }

}
```

You will need to make `FreeboxApiManager.hmacSha1` public to make this work.
Any of the classes you need to import should be

**Comment by lolodomo** [comment]
Thank you for the help, I did not imagine it was just one class to add.

File: bundles/org.openhab.binding.freebox/src/test/java/org/openhab/binding/freebox/internal/api/FreeboxApiManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+public class FreeboxApiManagerTest {
+
+    @Test
+    public void hmacSha1Test() throws Exception {
+        String expected = "25dad1bb5604321f12b755cc9d755d1480cf7989";
+        String actual = FreeboxApiManager.hmacSha1("Token1234", "Challenge");
+        Assert.assertEquals(expected, actual);
+    }
+
+}
```
Comment (cpmeister) [inline bundles/org.openhab.binding.freebox/src/test/java/org/openhab/binding/freebox/internal/api/FreeboxApiManagerTest.java:33]: Some of our build tools don't like EOF without a newline. So add a newline here.

**Comment by cpmeister** [comment]
> Thank you for the help, I did not imagine it was just one class to add.

Unit testing with JUnit is made to be as simple as possible with the idea that the easier it is the more likely unit tests would be written. More tests means more stable code, so it is a win-win for developers.

**Pull request merged**